src/wh_server_crypto.c

```diff
@@ -649,6 +647,79 @@ int wh_Server_HandleCryptoRequest(whServerContext* server,
         }
         break;
 #endif /* !WC_NO_RNG */
+#ifdef WOLFSSL_CMAC
+    case WC_ALGO_TYPE_CMAC:
+        /* in, out and key are after the fixed size fields */
+        in = (uint8_t*)(&packet->cmacReq + 1);
+        key = in + packet->cmacReq.inSz;
+        out = (uint8_t*)(&packet->cmacRes + 1);
+        /* do oneshot if all fields are present */
+        if (packet->cmacReq.inSz != 0 && packet->cmacReq.keySz != 0 &&
+            packet->cmacReq.outSz != 0) {
```

Comment:
Check alternately for keySz==0 but cmacReq.keyId !=NULL.  Then, read the key and invoke the oneshot.